backend/src/reservations/reservations.controller.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Put,
+  Req,
+  Param,
+  Body,
+  UsePipes,
+  Res,
+} from '@nestjs/common';
+import { ReservationsDto } from './reservations.dto';
+import { ReservationService } from './reservations.service';
+import { ValidationPipe } from 'validations/validation.pipe';
+import { Reservations } from './reservations.entity';
+
+@Controller('reservations')
+export class ReservationsController {
+  constructor(private readonly reservationsService: ReservationService) {}
+  @Get()
+  async findAll(@Res() response): Promise<Reservations[]> {
+    try {
+      const reservations = await this.reservationsService.findAll();
+      return response.status(200).send(reservations);
+    } catch (error) {
+      return response
+        .status(500)
+        .send(`Failed to get reservations: ${error.message}`);
+    }
+  }
+
+  @Post()
+  @UsePipes(new ValidationPipe())
+  async create(@Res() response, @Body() reservation: ReservationsDto) {
+    try {
+      const reservationDto = await this.reservationsService.createReservation(
+        reservation,
+      );
+      return response.status(201).send(reservation);
+    } catch (error) {
+      return response
+        .status(500)
+        .send(`Failed to create new reservation: ${error.message}`);
+    }
+  }
+
+  @Get(':id')
+  async findOne(@Res() response, @Param('id') id) {
+    try {
+      const reservation = await this.reservationsService.findOneById(id);
+      return response.status(200).send(reservation);
+    } catch (error) {
+      return response
+        .status(500)
+        .send(`Failed to get reservation: ${error.message}`);
```

Comment:
Here it might be a good idea to be a bit more specific with the different error codes you are sending.. eg. if there were no reservations found, you would return status(404), if the params were wrong you would return status (400) as bad request.. and so on.. status(500) should be only used when the application is not working as it should.

Reply:
Yes I agree to this. I will update accordingly. Thank you :)